src/overlays/actors/ovl_En_Dns/z_en_dns.c

```diff
@@ -27,74 +51,585 @@ const ActorInit En_Dns_InitVars = {
+            this->actor.flags &= ~0x1;
+            func_8013AED4(&this->unk_2C6, 0, 7);
+            this->unk_2C6 |= 0x80;
+            this->unk_1D8 = 0xFF;
+        }
+        phi_v1 = 1;
+    } else if (this->unk_2C6 & 0x80) {
+        this->actor.flags |= 1;
+        func_8013AED4(&this->unk_2C6, 3, 7);
+        this->unk_2C6 &= ~0x80;
+    }
+    return phi_v1;
+}
+
+s32 func_8092CC68(GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 pad[2];
+    s32 ret = false;
+    s16 bgId;
+
+    if (!func_801690CC(globalCtx) && (player->actor.bgCheckFlags & 1) && (player->transformation != PLAYER_FORM_DEKU)) {
+        bgId = player->actor.floorBgId;
+        if (func_800C99AC(&globalCtx->colCtx, player->actor.floorPoly, bgId) != 4) {
+            ret = true;
+        }
+    }
+
+    return ret;
+}
+
+s32 func_8092CCEC(EnDns* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 pad;
+    Vec3f sp3C = player->actor.world.pos;
+    Vec3f sp30 = this->actor.world.pos;
+    s16 sp2E;
+
+    Math_Vec3f_Copy(&sp30, &this->actor.world.pos);
+    Math_Vec3f_Copy(&sp3C, &player->actor.world.pos);
+    this->unk_2D6 = Math_Vec3f_Yaw(&D_801D15B0, &sp3C);
+    this->unk_2D4 = Math_Vec3f_Yaw(&D_801D15B0, &sp30);
+    this->unk_2EC = Math_Vec3f_DistXZ(&sp30, &D_801D15B0);
+    sp2E = Math_Vec3f_Yaw(&D_801D15B0, &sp3C);
+    sp2E -= Math_Vec3f_Yaw(&D_801D15B0, &sp30);
+    this->unk_2D8 = (Rand_ZeroOne() * 182.0f) + 182.0f;
+    this->unk_2D8 = (sp2E > 0) ? this->unk_2D8 : -this->unk_2D8;
+    this->unk_2D0 = 0x28;
+    this->actor.shape.shadowDraw = func_800B3FC0;
+    return 1;
+}
+
+s32 func_8092CE38(EnDns* this) {
+    static s32 D_8092DE00[] = { 8, 8, 9 };
+    s16 frame;
+    s32 pad;
+    Vec3f sp2C;
+    s32 ret = false;
+
+    if ((this->unk_2C6 & 0x200) || func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount)) {
+        func_8092C63C(this, D_8092DE00[this->unk_2D2]);
+        this->unk_2C6 &= ~0x200;
+        this->skelAnime.animCurrentFrame = 0.0f;
+        if (this->unk_2D2 == 2) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_NUTS_JUMP);
+        }
+        this->unk_2D2 += 1;
+        if (this->unk_2D2 >= 3) {
+            this->unk_2D2 = 0;
+        }
+        ret = true;
+    } else {
+        if (this->unk_2D2 == 0) {
+            if (func_801378B8(&this->skelAnime, 13.0f)) {
+                this->actor.world.rot.y = BINANG_ROT180(this->actor.world.rot.y);
+                this->unk_2E4 = 0.0f;
+                this->actor.shape.rot.y = this->actor.world.rot.y;
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_NUTS_JUMP);
+            } else if (this->skelAnime.animCurrentFrame < 13.0f) {
+                frame = this->skelAnime.animCurrentFrame;
+                this->actor.shape.rot.y = this->actor.world.rot.y;
+                frame *= 2520;
+                this->actor.shape.rot.y += frame;
+                this->unk_2E4 -= -3.0769231f;
```

Comment:
```suggestion
                this->unk_2E4 -= -(40.0f / 13.0f);
```

Reply:
Phew! I'm so glad I'm not the only one who does this (educated guess the fraction corresponding to a float constant). I second all of engineer's suggestions here.

Reply:
You can also just take floats and throw them into Wolfram Alpha. It'll give you the right fraction nearly every time without any effort :)

Reply:
The annoying thing is checking if it gives the right hex representation. We could do with a hex calculator that can do that.